shell/store/github.js

```diff
@@ -1,239 +1,95 @@
-
-  if ( updated && dayjs().diff(updated) <= 60 * 60 * 1000 ) {
-    return false;
-  }
-
-  return true;
-}
-
-function setCache(repos, scopes) {
-  if ( repos ) {
-    window.localStorage.setItem(GITHUB_REPOS, JSON.stringify(repos));
-    window.localStorage.setItem(GITHUB_REPOS + _DATE, (new Date()).toISOString());
-  }
-
-  if ( scopes ) {
-    window.localStorage.setItem(GITHUB_SCOPES, JSON.stringify(scopes));
-  }
-}
-
-function forgetCache() {
-  window.localStorage.removeItem(GITHUB_REPOS);
-  window.localStorage.removeItem(GITHUB_REPOS + _DATE);
-  window.localStorage.removeItem(GITHUB_SCOPES);
-}
-
-function proxifyUrl(url) {
-  // Strip off absolute links to github API
-  if ( url.startsWith(API_BASE) ) {
-    url = url.substr(API_BASE.length);
-  }
-
-  // Add our proxy prefix
-  url = `/v1/github/${ url.replace(/^\/+/, '') }`;
-
-  // Less pages please
-  addParam(url, 'per_page', 100);
-
-  return url;
-}
+export const GITHUB_BASE_API = 'https://api.github.com';
 
 export const state = function() {
   return {
-    repos:  [],
-    scopes: []
+    repos:    [],
+    commits:  [],
+    branches: [],
+    scopes:   [],
+    errors:   null,
   };
 };
 
-export const actions = {
-  async apiList({ commit, dispatch }, {
-    url = null,
-    single = false,
-    depaginate = true,
-    onPageFn = null,
-    objectKey = 'items',
-  } = {}) {
-    const out = [];
-
-    url = proxifyUrl(url);
-
-    while ( true ) {
-      console.log('Github Request:', url); // eslint-disable-line no-console
-      const res = await dispatch('rancher/request', { url }, { root: true });
-      const links = parseLinkHeader(res._headers['link']);
-
-      const scopes = res._headers['x-oauth-scopes'];
-
-      if ( scopes ) {
-        commit('setScopes', scopes.split(/\s*,\s*/));
-      }
-
-      if ( single ) {
-        return res;
-      }
-
-      addObjects(out, isArray(res) ? res : res[objectKey]);
-
-      if ( onPageFn ) {
-        onPageFn(out);
-      }
+export const FetchGithubAPI = async(endpoint) => {
+  try {
+    const response = await fetch(`${ GITHUB_BASE_API }/${ endpoint }`);
 
-      if ( depaginate && links.next ) {
-        url = proxifyUrl(links.next);
-      } else {
-        break;
-      }
+    if (!response.ok) {
+      return await response.json();
     }
 
-    return out;
-  },
-
-  forgetCache() {
-    forgetCache();
-  },
-
-  async fetchRecentRepos({ commit, dispatch }, { allowCache = true } = {}) {
-    const cachedScopes = getCachedScopes();
-
-    if ( cachedScopes ) {
-      commit('setScopes', cachedScopes);
+    return await response.json();
+  } catch (error) {
+    if (error.response) {
+      return error.response;
     }
+  }
+};
 
-    if ( allowCache && hasCached() ) {
-      const cached = getCachedRepos();
-
-      if ( cacheExpired() ) {
-        dispatch('fetchRecentRepos', { allowCache: false });
-      }
-
-      return cached;
+export const actions = {
+  // eslint-disable-next-line no-empty-pattern
```

Comment:
We generally have a pattern to use `ctx` as the first arg even if it's not used. That removes the requirement to disable linting for `{ }`

Reply:
sorry, wasn't quite clear, `ctx` is the whole object, so it would be `async apiList(ctx, {`